src/transformers/models/gpt_neo/modeling_gpt_neo.py

```diff
@@ -309,14 +309,6 @@ def forward(
             output_attentions=output_attentions,
         )
 
-        # cache the hidden_states instead of key_value_states
-        # for local attention layer
-        if self.attention_type == "local":
```

Comment:
@finetuneanon I'm curious about this change. Does this mean the current implementation of use_cache is incorrect? This seems to fix the problem in #11787.

Reply:
The current implementation of local attention is rather convoluted and requires use_cache to work differently for local attention layers. Think it's probably correct however. Due to the simplifications done in this PR, it is not necessary to handle local attention cache differently from regular cache anymore.